src/flake8_simplify/rules/ast_with.rs

```diff
@@ -1,22 +1,56 @@
-use rustpython_ast::{Stmt, StmtKind};
+use rustpython_ast::{Stmt, StmtKind, Withitem};
 
+use crate::ast::helpers::first_colon_range;
 use crate::ast::types::Range;
 use crate::checkers::ast::Checker;
 use crate::registry::Diagnostic;
 use crate::violations;
 
-/// SIM117
-pub fn multiple_with_statements(checker: &mut Checker, stmt: &Stmt) {
-    let StmtKind::With { body, .. } = &stmt.node else {
-        return;
-    };
+fn find_nested_with(body: &[Stmt]) -> Option<(&Vec<Withitem>, &Vec<Stmt>)> {
     if body.len() != 1 {
-        return;
+        return None;
+    }
+    let StmtKind::With { items, body, .. } = &body[0].node else {
+        return None
+    };
+    find_nested_with(body).or(Some((items, body)))
```

Comment:
I think the following would be more idiomatic:

```rust
    let [StmtKind::With { items, body, .. }] = body else { return; };
```

Reply:
```
    let [StmtKind::With { items, body, .. }] = body else { return; };
```

Does this work? `body` is `&[Stmt]`, not `&[StmtKind]`?

Reply:
I think we could do:

```diff
 fn find_last_with(body: &[Stmt]) -> Option<(&Vec<Withitem>, &Vec<Stmt>)> {
-    if body.len() != 1 {
+    let [stmt] = body else {
         return None;
-    }
-    let StmtKind::With { items, body, .. } = &body[0].node else {
+    };
+    let StmtKind::With { items, body, .. } = &stmt.node else {
         return None
     };
     find_last_with(body).or(Some((items, body)))
     find_last_with(body).or(Some((items, body)))
```

but not sure if it't worth it

Reply:
Ah right I missed the additional `Locate` nesting ... I think sth like the following should work:

    let [Locate { node: StmtKind::With { items, body, .. }, ..}] = body else { return; };